### PR TITLE
refactor(core): type cleaning for render3 code base.

### PR DESCRIPTION
**Pull request by JeanMeche**
See individual commits, this PR improves the typings in the R3 code base. 

There should be no functionnal change. 

## PR Type
What kind of change does this PR introduce?

- [x] Refactoring (no functional changes, no api changes)

### CODE REVIEW

File: packages/core/src/render3/assert.ts [hunk 1]
```diff
@@ -27,10 +27,7 @@ export function assertTNodeForLView(tNode: TNode, lView: LView) {
 
 export function assertTNodeForTView(tNode: TNode, tView: TView) {
   assertTNode(tNode);
-  tNode.hasOwnProperty('tView_') &&
-      assertEqual(
-          (tNode as any as {tView_: TView}).tView_, tView,
```
Comment (JeanMeche) [inline packages/core/src/render3/assert.ts:32]: `tView_` does not exist anymore.

**Comment by JeanMeche** [comment]
I was thinking about adding some typings to the definitions (instead on `unknown` currently) to improve type assertions in the test. 

Screenshot 2023-03-13 at 00 21 19


What do you think ?

**Comment by JeanMeche** [comment]
I should probably split this in multiple PRs to ease the review.

**Comment by pkozlowski-opensource** [comment]
> I should probably split this in multiple PRs to ease the review.

Yes :-)

**Pull request closed without merging**